--- a/src/file.h
+++ b/src/file.h
@@ -96,7 +96,7 @@
 #define ENABLE_CONDITIONALS
 
 #ifndef MAGIC
-#define MAGIC "/etc/magic"
+#define MAGIC "/data/data/com.termux/files/usr/etc/magic"
 #endif
 
 #if defined(__EMX__) || defined (WIN32)

--- a/src/file.c
+++ b/src/file.c
@@ -327,7 +327,7 @@
                case 'v':
                        if (magicfile == NULL)
                                magicfile = magic_getpath(magicfile, action);
-                       (void)fprintf(stdout, "%s-%s\n", file_getprogname(),
+                       (void)fprintf(stdout, "%s v%s\nby PhateValleyman\nJonas.Ned@outlook.com\n", file_getprogname(),
                            VERSION);
                        (void)fprintf(stdout, "magic file from %s\n",
                            magicfile);

--- a/src/compress.c
+++ b/src/compress.c
@@ -465,7 +465,7 @@
                (void)strlcat(buf, "/", sizeof(buf));
        (void)strlcat(buf, "file.XXXXXX", sizeof(buf));
 #else
-       (void)strlcpy(buf, "/tmp/file.XXXXXX", sizeof(buf));
+       (void)strlcpy(buf, "/data/data/com.termux/files/usr/tmp/file.XXXXXX", sizeof buf);
 #endif
 #ifndef HAVE_MKSTEMP
        {
