--- a/pkg/cli/version.go
+++ b/pkg/cli/version.go
@@ -45,7 +45,7 @@
 			if version.full {
 				_, err = fmt.Fprintf(cmd.OutOrStdout(), "version: %s\ncommit: %s\ncommitDate: %s\n", buildinfo.Version(), buildinfo.Commit(), buildinfo.CommitDate())
 			} else {
-				_, err = fmt.Fprintf(cmd.OutOrStdout(), "%s\n", buildinfo.Version())
+				_, err = fmt.Fprintf(cmd.OutOrStdout(), "sgpt %s\nby PhateValleyman\nJonas.Ned@outlook.com\n", buildinfo.Version())
 			}
 			return err
 		},
