diff -uNr sgpt/pkg/cli/version.go sgpt.mod/pkg/cli/version.go
--- version.go	2023-12-12 23:20:39.400001096 +0100
+++ pkg/cli/version.go	2023-12-12 23:09:25.130000775 +0100
@@ -45,7 +45,7 @@
 			if version.full {
 				_, err = fmt.Fprintf(cmd.OutOrStdout(), "version: %s\ncommit: %s\ncommitDate: %s\n", buildinfo.Version(), buildinfo.Commit(), buildinfo.CommitDate())
 			} else {
-				_, err = fmt.Fprintf(cmd.OutOrStdout(), "%s\n", buildinfo.Version())
+				_, err = fmt.Fprintf(cmd.OutOrStdout(), "sgpt %s\nby PhateValleyman\nJonas.Ned@outlook.com\n", buildinfo.Version())
 			}
 			return err
 		},
