--- a/tree.c
+++ b/tree.c
@@ -18,11 +18,11 @@
 
 #include "tree.h"
 
-char *version = "$Version: $ tree v2.2.1 %s 1996 - 2024 by Steve Baker, Thomas Moore, Francesc Rocher, Florian Sesser, Kyosuke Tokoro $";
-char *hversion= "\t\t tree v2.2.1 %s 1996 - 2024 by Steve Baker and Thomas Moore <br>\n"
-		"\t\t HTML output hacked and copyleft %s 1998 by Francesc Rocher <br>\n"
-		"\t\t JSON output hacked and copyleft %s 2014 by Florian Sesser <br>\n"
-		"\t\t Charsets / OS/2 support %s 2001 by Kyosuke Tokoro\n";
+char *version = "$Version: $ tree v2.2.1-1\nby PhateValleyman\nJonas.Ned@outlook.com $";
+char *hversion= "\t\t tree v2.2.1-1\nby PhateValleyman\nJonas.Ned@outlook.com <br>\n"
+		"\t\t HTML output hacked and copyleft %s 1998 by PhateValleyman <br>\n"
+		"\t\t JSON output hacked and copyleft %s 2014 by PhateValleyman <br>\n"
+		"\t\t Charsets / OS/2 support %s 2001 by PhateValleyman\n";
 
 /* Globals */
 bool dflag, lflag, pflag, sflag, Fflag, aflag, fflag, uflag, gflag;
